### PR TITLE
Implement request resume review for student

### LINKED ISSUE #109: Integrate resume review page with UI and redux
Depends on #99

### LINKED ISSUE #188: Integrate post resume review with front end

### PR DESCRIPTION
**Pull request by armiantos**
# Overview

Allow students to request resume review by uploading a resume to the server

# Changes

* Add `filePromise` prop to PDFViewer
* Add UploadResume component
* Nest `NoResumes` inside `ResumeList`
* Create UploadResume slice
* Integrate `initiateResumeReview` with redux

# Questions & Notes

* Do we want cancel to go back to the 'Choose file' page or go back to the resume review list?
* I want to refactor the page titles as a component in some time in the future otherwise justify center will move the title's relative height if contents are too tall

# Issue tracking

Closes #188 
Closes #109 

# Testing & Demo

* Set `currentRole` to `student` in `UserSlice` and run locally
* Also refer to unit tests

https://user-images.githubusercontent.com/43416725/129634811-1ea73171-eab2-4f0d-b8e7-9e4c40ebe8a8.mov

### CODE REVIEW

**Comment by calebschoepp** [comment]
FWIW I think we should remove this

> All resumes are removed from our database one week after review

This is adding extra code/bugs for us to deal with us and I really don't think people care.

**Comment by armiantos** [comment]
> FWIW I think we should remove this
> 
> > All resumes are removed from our database one week after review
> 
> This is adding extra code/bugs for us to deal with us and I really don't think people care.

Won't we exceed our storage quota in S3 though?

File: www/src/routes/student/ResumeReview.tsx [hunk 1]
```diff
@@ -1,46 +1,28 @@
...
-                    <Grid container item xs={12} justify='center'>
-                        <Typography>You have no resumes submitted</Typography>
-                    </Grid>
-                    <Grid container item xs={12} justify='center'>
-                        <Button variant='contained' color='primary'>
-                            Submit resume
-                        </Button>
-                    </Grid>
-                </Grid>
+                {isUploading ? <UploadResume /> : <ResumeList />}
```
Comment (calebschoepp) [inline www/src/routes/student/ResumeReview.tsx:25]: Nice and clean the way you split these out.

File: www/src/routes/student/resumeReview/UploadResume.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+import { resetUploadResume, setDocument } from '../../../redux/substores/student/slices/uploadResumeSlice';
+import { useStudentDispatch, useStudentSelector } from '../../../redux/substores/student/studentHooks';
+import { StudentDispatch } from '../../../redux/substores/student/studentStore';
+import initiateResumeReview from '../../../redux/substores/student/thunks/initiateResumeReview';
+import { arrayBufferToBase64, base64ToArrayBuffer } from '../../../util/helpers';
+
+const handleOnFileSelected = async (dispatch: StudentDispatch, files?: FileList | null) => {
+    // TODO: Improve error handling
+    if (files === undefined || files === null) {
+        alert('No file selected');
```
Comment (calebschoepp) [inline www/src/routes/student/resumeReview/UploadResume.tsx:18]: I agree with using alerts. Simple and straightforward for the limited time we have.

**Review by calebschoepp — APPROVED** [review]
LGTM! :rocket:

File: www/src/routes/student/ResumeReview.tsx (hunk 1 above)
Comment (armiantos) [reply www/src/routes/student/ResumeReview.tsx:25]: Thanks! 😄

File: www/src/routes/student/resumeReview/UploadResume.tsx (hunk 2 above)
Comment (armiantos) [reply www/src/routes/student/resumeReview/UploadResume.tsx:18]: Yea, pop ups are will kick us sooner than later 😂

**Pull request merged**